boards/BMS/Src/Io/Io_LTC6813/Io_NewLTC6813.c

```diff
@@ -0,0 +1,318 @@
+#include "Io_NewLTC6813.h"
+#include "Io_NewCellVoltages.h"
+#include "App_Accumulator.h"
+#include "App_SharedMacros.h"
+
+// Time that a SPI transaction should wait for until until an error is returned
+#define SPI_TIMEOUT_MS (10U)
+
+// Max number of commands to check for completed ADC conversions
+#define MAX_NUM_ADC_COMPLETE_CHECKS (10U)
+
+// Command used to start/stop cell discharge
+#define MUTE (0x0028U)
+#define UNMUTE (0x0029U)
+
+// Command used to write to configuration registers
+#define WRCFGA (0x0001U)
+#define WRCFGB (0x0024U)
+
+// Command used to poll ADC conversions
+#define PLADC (0x0714U)
+#define PLADC_RX_SIZE (1U)
+#define ADC_CONV_INCOMPLETE (0xFFU)
+
+// clang-format off
+// Default configurations for CFGRA and CFGRB
+#define VUV (0x4E1U)
+#define VOV (0x8CAU)
+#define ADCOPT (1U)
+#define REFON (0U << 2U)
+#define DTEN (0U << 1U)
+#define ENABLE_ALL_CFGRA_GPIO (0x001FU << 3U)
+#define ENABLE_ALL_CFGRB_GPIO (0x000FU)
+
+#define DEFAULT_CFGRA_CONFIG                                                           \
+    {                                                                                  \
+        [REG_GROUP_BYTE_0] = (uint8_t)(ENABLE_ALL_CFGRA_GPIO | REFON | DTEN | ADCOPT), \
+        [REG_GROUP_BYTE_1] = (uint8_t)VUV,                                             \
+        [REG_GROUP_BYTE_2] = (uint8_t)((VOV & 0xF) << 4) + (VUV >> 8),                 \
+        [REG_GROUP_BYTE_3] = (uint8_t)(VOV >> 4), [REG_GROUP_BYTE_4] = 0x0U,           \
+        [REG_GROUP_BYTE_5] = 0x0U,                                                     \
+        [REG_GROUP_PEC0] = 0U,                                                         \
+        [REG_GROUP_PEC1] = 0U,                                                         \
+    }
+
+#define DEFAULT_CFGRB_CONFIG                        \
+    {                                               \
+        [REG_GROUP_BYTE_0] = ENABLE_ALL_CFGRB_GPIO, \
+        [REG_GROUP_BYTE_1] = 0U,                    \
+        [REG_GROUP_BYTE_2] = 0U,                    \
+        [REG_GROUP_BYTE_3] = 0U,                    \
+        [REG_GROUP_BYTE_4] = 0U,                    \
+        [REG_GROUP_BYTE_5] = 0U,                    \
+        [REG_GROUP_PEC0] = 0U,                      \
+        [REG_GROUP_PEC1] = 0U,                      \
+    }
+// clang-format on
+
+// Macros used to set DCC bits in the configuration register
+#define SET_ALL_DCC_BITS (0xFFFFU)
+#define SET_MIN_CELL_DCC_BIT(index) ((uint16_t)(1U << (index)))
+#define SET_CFGRA4_DCC_BITS(dcc_bits) (0xFFU & (uint8_t)(dcc_bits))
+#define SET_CFGRA5_DCC_BITS(dcc_bits) (0x0FU & (uint8_t)((dcc_bits) >> 8U))
+#define SET_CFGRB0_DCC_BITS(dcc_bits) (0xF0U & (uint8_t)((dcc_bits) >> 8U))
+
+#define PEC15_LUT_SIZE (256U)
+
+enum ConfigurationRegister
+{
+    CONFIG_REG_A = 0,
+    CONFIG_REG_B,
+    NUM_OF_CFG_REGISTERS,
+};
+
+// TODO: fix this
+extern struct SharedSpi *ltc6813_spi;
+// struct SharedSpi *       ltc6813_spi = NULL;
```

Comment:
this extern was conflicting with the one for the current ltc6813_spi being initialized, leaving as extern for now